include/elf_parser.hpp

```diff
@@ -0,0 +1,191 @@
+#pragma once
+#include <bitset>
+#include <err.h>
+#include <fcntl.h>
+#include <gelf.h>
+#include <libelf.h>
+#include <print>
+#include <stdlib.h>
+#include <string>
+#include <sysexits.h>
+#include <unistd.h>
+
+class ElfParser
+{
+  public:
+    ElfParser(char** t_file_ptr)
+      : m_file(t_file_ptr){};
+    void openElf();
+    void printEhdr();
+    void printPhdr();
+    void printShdr();
+    //void printSym();
+    void getSection(std::string section);
+    void closeElf();
+
+  private:
+    int m_elf_class{ 0 };
+    int m_fd, m_fbin;
+    size_t m_n, m_shstrndx;
+
+    char** m_file;
+
+    char* m_name;
+    Elf_Scn* m_scn;
+    Elf_Data* m_data;
+
+    Elf* m_e;
+    GElf_Ehdr m_ehdr;
+    GElf_Phdr m_phdr;
+    GElf_Shdr m_shdr;
+    //Gelf_Sym m_sym;
+};
+
+void ElfParser::openElf()
```

Comment:
```suggestion
inline void ElfParser::openElf()
```

You need inline to allow functions to exist in headers.

Reply:
But you should throw this into a source file.